Add support of rcar spider ca55 board

xakep-amatop opened the pull request.

xakep-amatop committed:
boards: arm64: add support of Renesas Spider S4 A55 board

Add support of 'rcar_spider_s4/r8a779f0/a55' board: minimal dts
and configuration.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
snippets: xen_dom0: add support of rcar_spider_ca55

Add support of rcar_spider_ca55 to xen_dom0 snippet.

Disable CONFIG_UART_INTERRUPT_DRIVEN for xen_dom0, because
xen dom0 consoleio doesn't support interrupt driven mode.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

nordicjm commented on soc/renesas/rcar/rcar_gen4/Kconfig:
```diff
@@ -1,10 +1,23 @@
 # SPDX-License-Identifier: Apache-2.0
 
-config SOC_SERIES_RCAR_GEN4
+config SOC_SERIES_RCAR_GEN4_R52
 	bool
 	select ARM
 	select CPU_CORTEX_R52
 	select GIC_SINGLE_SECURITY_STATE
 	select CLOCK_CONTROL_RCAR_CPG_MSSR if CLOCK_CONTROL
 	select ARM_ARCH_TIMER
+
+config SOC_SERIES_RCAR_GEN4_A55
+	bool
+	select ARM64
+	select CPU_CORTEX_A55
+	select CLOCK_CONTROL_RCAR_CPG_MSSR if CLOCK_CONTROL
+	select ARM_ARCH_TIMER
+
```
That needs fixing

xakep-amatop replied:
changed

xakep-amatop commented on drivers/pinctrl/renesas/rcar/CMakeLists.txt:
```diff
@@ -8,4 +8,7 @@ if (CONFIG_SOC_R8A77951_R7 OR CONFIG_SOC_R8A77951_A57)
 
 zephyr_library_sources_ifdef(CONFIG_SOC_R8A77961 pfc_r8a77961.c)
-zephyr_library_sources_ifdef(CONFIG_SOC_R8A779F0 pfc_r8a779f0.c)
+
```
done

nordicjm commented on boards/renesas/rcar_spider_s4/Kconfig.rcar_spider_s4:
```diff
@@ -2,4 +2,8 @@
+	select SOC_R8A779F0_R52 if BOARD_RCAR_SPIDER_S4_R8A779F0_R52
+	select SOC_R8A779F0_A55 if BOARD_RCAR_SPIDER_S4_R8A779F0_A55
+
+config BUILD_OUTPUT_BIN
+	default y if BOARD_RCAR_SPIDER_S4_R8A779F0_A55
```
`Kconfig`

xakep-amatop replied:
moved to `Kconfig.defconfig`

nordicjm reviewed (changes requested):
Minor change then looks good

aaillet reviewed (dismissed):
LGTM, thanks for this work

xakep-amatop commented:
Can these changes be merged?

xakep-amatop commented:
Rebased changes and deleted changes from `sbs_charger` and `fuel_gauge` tests after fixing the issue (https://github.com/zephyrproject-rtos/zephyr/issues/71870) with tests filtering.
